go/def.bzl

```diff
@@ -389,9 +402,25 @@ def emit_go_link_action(ctx, importmap, transitive_libs, cgo_deps, lib,
   # Avoided -s on OSX but but it requires dsymutil to be on $PATH.
   # TODO(yugui) Remove this workaround once rules_go stops supporting XCode 7.2
   # or earlier.
-  cmds += ["export PATH=$PATH:/usr/bin"]
   cmds += [
+    "export PATH=$PATH:/usr/bin",
     "export GOROOT=$(pwd)/" + ctx.file.go_tool.dirname + "/..",
+    "STAMP_XDEFS=()",
+  ]
+
+  stamp_input = []
+  if ctx.attr.linkstamp:
+    stamp_input = [ctx.info_file, ctx.version_file]
+    for f in stamp_input:
+      cmds += [
+          "while read -r stamp || [[ -n $stamp ]]; do",
+          # reads workspace status file, converting "KEY value" lines to "-X KEY=$linkstamp.value" arguments to the go linker.
+          "if echo $stamp | grep -qe '^.*[[:space:]].*$'; then",
+          "STAMP_XDEFS+=(-X \"$(echo \"%s.${stamp}\" | tr ' ' '=')\");" % ctx.attr.linkstamp,
```

Comment:
Do we need the grep? It matches any line with whitespace, right? I'd imagine in some cases, the value for some key would be empty, and a line might not have any space. Maybe use sed instead?

    echo $stamp | sed -e 's%^\([^ ]\+\) \?\(.*\)$%-Xa/b.\1=\2%'